### DISCUSSION TITLE
API-driven CMS with Astro Paper v2

### DISCUSSION THREAD

**Discussion by 66Leo66** [satnaing/astro-paper#39] [Q&A]
I'd like to use directus cms with this theme. I've read the relevant issue but that seemed to be based on v1 version of the theme. My questions are:

1. What's the "most painless" way to connect to a CMS? Say, can I fetch data and re-organize them to fit `getCollection()` format so Astro Paper's internal logics can remain unchanged?
2. How can I directly use HTML fetched from CMS?

Thanks!

**Comment by satnaing** [comment]
Hello @66Leo66,
First of all, a good discussion and questions. 
I'm also planning to update a blog post about connecting AstroPaper to a headless CMS. 
But for that, first, I have to try developing a compatible schema for the theme and experimenting CMSes.

Before this, you can try fetching CMS data as I mentioned in [this issue](https://github.com/satnaing/astro-paper/issues/19#issuecomment-1377435607).
